nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SetSNMP.java

```diff
@@ -16,209 +16,127 @@
-                        this.getLogger().error("Failed while executing SNMP Set [{}] via " + this.targetResource + ". Error = {}", new Object[]{response.getRequest().getVariableBindings(), error});
-                    }
-                } catch (IOException e) {
-                    processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                    this.getLogger().error("Failed while executing SNMP Set via " + this.targetResource, e);
-                    context.yield();
-                }
+                processPdu(context, processSession, flowFile, pdu);
             } else {
                 processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
-                this.getLogger().warn("No attributes found in the FlowFile to perform SNMP Set");
+                getLogger().warn("No attributes found in the FlowFile to perform SNMP Set");
             }
         }
     }
 
-    /**
-     * Method to construct {@link VariableBinding} based on {@link FlowFile}
-     * attributes in order to update the {@link PDU} that is going to be sent to
-     * the SNMP Agent.
-     * @param pdu {@link PDU} to be sent
-     * @param attributes {@link FlowFile} attributes
-     * @return true if at least one {@link VariableBinding} has been created, false otherwise
-     */
-    private boolean addVariables(PDU pdu, Map<String, String> attributes) {
-        boolean result = false;
-        for (Entry<String, String> attributeEntry : attributes.entrySet()) {
-            if (attributeEntry.getKey().startsWith(SNMPUtils.SNMP_PROP_PREFIX)) {
-                String[] splits = attributeEntry.getKey().split("\\" + SNMPUtils.SNMP_PROP_DELIMITER);
-                String snmpPropName = splits[1];
-                String snmpPropValue = attributeEntry.getValue();
-                if(SNMPUtils.OID_PATTERN.matcher(snmpPropName).matches()) {
-                    Variable var = null;
-                    if (splits.length == 2) { // no SMI syntax defined
-                        var = new OctetString(snmpPropValue);
-                    } else {
-                        int smiSyntax = Integer.valueOf(splits[2]);
-                        var = this.stringToVariable(snmpPropValue, smiSyntax);
-                    }
-                    if(var != null) {
-                        VariableBinding varBind = new VariableBinding(new OID(snmpPropName), var);
-                        pdu.add(varBind);
-                        result = true;
-                    }
-                }
+    private void processPdu(ProcessContext context, ProcessSession processSession, FlowFile flowFile, PDU pdu) {
+        try {
+            ResponseEvent response = snmpSetter.set(pdu);
+            if (response.getResponse() == null) {
+                processSession.transfer(processSession.penalize(flowFile), REL_FAILURE);
+                getLogger().error("Set request timed out or parameters are incorrect.");
```

Comment:
Logging FlowFile or some identifying attributes would be helpful for troubleshooting.